File: tests/solver_integration/python_solver/import_data.py
import CoSimIO

def cosimio_check_equal(a, b):
    assert a == b

# Connection Settings
settings = CoSimIO.ConnectionSettings()
settings.SetString("connection_name", "test_connection") # This must be unique for each connection between two solvers
settings.SetString("solver_name", "my_solver") # Not to be confused with the connection name. 
settings.SetInt("echo_level", 1)
settings.SetString("solver_version", "1.25")

# Connecting
return_info = CoSimIO.Connect(settings)
cosimio_check_equal(return_info.GetInt("connection_status"), CoSimIO.ConnectionStatus.Connected)

# Importing data
info = CoSimIO.Info()
info.SetString("identifier", "vector_of_pi")
info.SetString("connection_name", "test_connection")
return_info, received_data = CoSimIO.ImportData(info)

cosimio_check_equal(len(received_data), 4)

for value in received_data:
    cosimio_check_equal(value, 3.14)


# Disconnecting
return_info = CoSimIO.Disconnect(settings)
cosimio_check_equal(return_info.GetInt("connection_status"), CoSimIO.ConnectionStatus.Disconnected)

File: tests/solver_integration/c/import_data.c
//     ______     _____ _           ________
//    / ____/___ / ___/(_)___ ___  /  _/ __ |
//   / /   / __ \\__ \/ / __ `__ \ / // / / /
//  / /___/ /_/ /__/ / / / / / / // // /_/ /
//  \____/\____/____/_/_/ /_/ /_/___/\____/
//  Kratos CoSimulationApplication
//
//  License:         BSD License, see license.txt
//
//  Main authors:    Pooyan Dadvand
//

// CoSimulation includes
#include "c/co_sim_io_c.h"

#define COSIMIO_CHECK_EQUAL(a, b)                                \
    if (a != b) {                                                \
        printf("in line %d : %d is not equalt to %d\n", __LINE__ , a, b); \
        return 1;                                                \
    }

int main()
{
    // CoSimIO::Info settings;
    // settings.Set("connection_name", "test_connection"); // This must be unique for each connection between two solvers
    // settings.Set("solver_name", "solver_1"); // Not to be confused with the connection name.
    // settings.Set("echo_level", 1);
    // settings.Set("solver_version", "1.25");

    // auto info = CoSimIO::Connect(settings);
    // COSIMIO_CHECK_EQUAL(info.Get<int>("connection_status"), CoSimIO::ConnectionStatus::Connected);

    // std::vector<double> receive_data;
    // info.Clear();
    // info.Set("identifier", "vector_of_pi");
    // info.Set("connection_name", "test_connection");
    // info = CoSimIO::ImportData(info, receive_data);

    // for(auto& value : receive_data)
    //     COSIMIO_CHECK_EQUAL(value, 3.14);

    // info = CoSimIO::Disconnect(settings); // disconnect afterwards
    // COSIMIO_CHECK_EQUAL(info.Get<int>("connection_status"), CoSimIO::ConnectionStatus::Disconnected);

    return 0;
}